cmd/deployment.go

```diff
@@ -0,0 +1,187 @@
+	defaultState := choiceflag.NewChoiceFlag(states)
+	_ = defaultState.Set("success")
+	flags.VarP(defaultState, "state", "s", "deployment state")
+
+	flags.BoolP("transient", "T", false, "transient environment")
+	flags.BoolP("production", "P", false, "production environment")
+	flags.String("description", "", "deployment description")
+	flags.String("environment-url", "", "environment URL")
+
+	addDryRunFlag(flags)
+
+	flags.SetNormalizeFunc(normalizeFlags)
+	flags.SortFlags = false
+
+	return cmd
+}
+
+func runDeploymentCmd(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+
+	environment := viper.GetString("environment")
+
+	if len(args) == 1 {
+		environment = args[0]
+	}
+
+	if environment == "" {
+		return fmt.Errorf("environment is required")
+	}
+
+	state := viper.GetString("state")
+
+	dryRun := viper.GetBool("dry-run")
+
+	repo := remote.Repo{
+		Owner: viper.GetString("owner"),
+		Name:  viper.GetString("repo"),
+	}
+
+	client, err := remote.NewClient(ctx, &repo)
+	if err != nil {
+		return fmt.Errorf("NewClient(%s): %w", repo, err)
+	}
+
+	commitish := viper.GetString("commitish")
+	var targetSha string
+	if commitish != "" {
+		targetSha, err = client.ResolveCommitish(commitish)
+		if err != nil {
+			return fmt.Errorf("ResolveCommitish(%s, %s): %w", repo, commitish, err)
+		}
+		if targetSha == "" {
+			return fmt.Errorf("commitish %q not found", commitish)
+		}
+	} else {
+		repoInfo, err := client.GetRepositoryInfo("")
+		if err != nil {
+			return fmt.Errorf("GetRepositoryInfo(%s): %w", repo, err)
+		}
+		commitish = repoInfo.DefaultBranch.Name
+		targetSha = string(repoInfo.DefaultBranch.Commit)
+	}
+
+	output := &DeploymentOutput{
+		Environment: environment,
+		Commitish:   commitish,
+		SHA:         targetSha,
+		State:       state,
+		URL:         client.GetCommitURL(targetSha),
+	}
+
+	// Check if deployment already exists
+	var deployment *remote.DeploymentInfo
+	var deploymentID string = "1" // Default for dry-run mode
+
+	if !dryRun {
+		deployments, err := client.ListDeploymentsV3(targetSha, environment)
+		if err != nil {
+			output.SetError(fmt.Errorf("listing deployments: %w", err))
+			return cmdOutput(cmd, output)
+		}
+
+		if len(deployments) > 0 {
+			// Use existing deployment
+			deployment = &deployments[0]
+			deploymentID = string(deployment.ID)
+			log.Infof("using existing deployment: %s", deploymentID)
+		} else {
+			// Create new deployment
+			transient := viper.GetBool("transient")
+			production := viper.GetBool("production")
+			description := viper.GetString("description")
+
+			log.Infof("creating deployment for %s in %s environment", commitish, environment)
+			deployment, err = client.CreateDeploymentV3(commitish, environment, description, transient, production)
+			if err != nil {
+				output.SetError(fmt.Errorf("creating deployment: %w", err))
+				return cmdOutput(cmd, output)
+			}
+
+			deploymentID = string(deployment.ID)
```

Comment:
No need to type convert:
```suggestion
			deploymentID = deployment.ID
```